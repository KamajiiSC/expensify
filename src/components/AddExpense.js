import React from 'react';

const AddExpense = () => (
  <div>
    for AddExpense
  </div>
);

export default AddExpense;