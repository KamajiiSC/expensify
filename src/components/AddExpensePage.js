import React from 'react';

const AddExpensePage = () => (
  <div>
    for AddExpense
  </div>
);

export default AddExpensePage;