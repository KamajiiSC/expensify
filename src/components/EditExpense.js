import React from 'react';

const EditExpense = () => (
  <div>
    Edit your expenses
  </div>
);

export default EditExpense;