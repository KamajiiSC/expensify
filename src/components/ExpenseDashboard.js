import React from 'react';

const ExpenseDashboard = () => (
  <div>
    Something to show
  </div>
);

export default ExpenseDashboard;