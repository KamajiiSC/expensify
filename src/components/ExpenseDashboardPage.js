import React from 'react';

const ExpenseDashboardPage = () => (
  <div>
    Something to show
  </div>
);

export default ExpenseDashboardPage;