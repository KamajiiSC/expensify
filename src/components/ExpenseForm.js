import React from 'react';

export default class ExpenseForm extends React.Component {
  state = {
    description: ''
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onTextAreaChange = (e) => {
    const area = e.target.value;
    this.setState(() => ({area}));
  };
  render() {
    return (
      <div>
        <form>
          <input 
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input 
            type="number"
            placeholder="Amount"
          />
          <textarea
          placeholder="Add a nnote for your expense(optional"
          onChange={this.onTextAreaChange}
          >
          </textarea>
          <button>Add Expense</button>
        </form>
      </div>
    )
  }
}