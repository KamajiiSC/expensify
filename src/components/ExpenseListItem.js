import React from 'react';

const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <h2>{description}</h2>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
  </div>
);

export default ExpenseListItem;