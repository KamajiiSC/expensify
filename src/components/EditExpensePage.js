import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm'
import { removeExpense, addExpense } from '../actions/expenses';
import { editExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  render() {
    return(
      <div>
        <ExpenseForm 
          expense={this.props.expense}
          onSubmit={(expense) => {
            this.props.dispatch(editExpense(this.props.expense.id, expense));
            this.props.history.push('/');
          }}
        />
        <button 
          onClick={() => {
            this.props.dispatch(removeExpense({ id: this.props.expense.id }));
            this.props.history.push('/');
          }}
          >Remove
        </button>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => {
  return{ 
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: () => dispatch(editExpense(expense)),
  removeExpense: () => dispatch(removeExpense(expense))
});

export default connect(mapDispatchToProps)(EditExpensePage);