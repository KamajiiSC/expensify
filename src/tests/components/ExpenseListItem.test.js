import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';