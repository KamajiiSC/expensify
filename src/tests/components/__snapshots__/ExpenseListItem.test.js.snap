// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an ExpenseListItem with data from fixture 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h2>
      rent
    </h2>
  </Link>
  <p>
    Amount: 
    19500
  </p>
  <p>
    Created at: 
    -345600000
  </p>
</div>
`;
